### DISCUSSION TITLE
[SDK]Error: Failed to ChangePubKey authorized by ECDSA.: Account does not exist in the zkSync network

### DISCUSSION THREAD

**Discussion by herryho** [zkSync-Community-Hub/zksync-developers#15] [SDKs] [answered]
### Environment

Mainnet

### Acknowledgement

- [X] I understand that issues are expected with the Python and Java SDK as they are being updated.

### Issue Description

I have an account（0x26b99F99aEF1230f9F691a3677d711639e70443B） which has already conducted some transactions in zksync through metamask. However, when I try to setSigningKey for the account by code, it prompted “Error: Failed to ChangePubKey authorized by ECDSA.: Account does not exist in the zkSync network”. 
After testing, I found: await syncWallet.getAccountId() == undefined. That's the problem. 
The code worked for some other accounts. So it is not the problem of the code.

### Expected Behavior

 I can set signing keys for the account through ChangePubKey successfully.

### Code Example

```
      const eth_endpoint = process.env.ETH_ENDPOINT;
      const syncProvider = await zksync.getDefaultProvider('mainnet');
      const ethersProvider = new ethers.providers.JsonRpcProvider(eth_endpoint);
      const ethSigner = new ethers.Wallet(privateKey, ethersProvider);
      this._syncWallet = await zksync.Wallet.fromEthSigner(
        ethSigner,
        syncProvider,
      );
     
          if (!(await syncWallet.isSigningKeySet())) {
        if ((await syncWallet.getAccountId()) == undefined) {
          throw new Error('Unknown account');
        }

        const changePubkey = await syncWallet.setSigningKey({
          feeToken: 'ETH',
          ethAuthType: 'ECDSA',
        });

        await changePubkey.awaitReceipt();
      }
```

**Comment by herryho** [comment]
@bxpana

**Reply by bxpana** [reply, marked as answer]
It sounds like you are using zkSync Lite if you are getting the `ChangePubKey` error <https://docs.zksync.io/api/sdk/js/accounts/#sign-zksync-changepubkey>. 

Are you trying to build on zkSync Lite or zkSync Era? Also which docs are you using to help you?

**Reply by herryho** [reply]
You are right! I wasn't aware that the changepubkey function is specific for zksync lite. I was very curious that why there are two js sdk. So I know now, "zksync" is for zksync lite, while "zksync-web3" is for zksync era.

Problem solved!

**Comment by newtondav** [comment]
It seems that the account in question may not be properly registered on the zkSync network. Ensure that the account has been properly created and funded on zkSync before attempting to set the signing key. You may need to investigate further to determine why the account does not have an associated ID on the zkSync network.

**Comment by fullertasha** [comment]
The error "Error: Failed to ChangePubKey authorized by ECDSA.: Account does not exist in the zkSync network" indicates that the account you are trying to set signing keys for does not exist in the zkSync network. This could be due to various reasons such as the account not being registered or synced properly. Ensure that the account has been properly registered and synced with the zkSync network before attempting to set signing keys.
